clients/fides-js/src/integrations/gtm.ts

```diff
@@ -10,35 +11,71 @@ declare global {
 /**
  * Defines the structure of the Fides variable pushed to the GTM data layer
  */
-type FidesVariable = FidesEvent["detail"];
+type FidesVariable = Omit<FidesEvent["detail"], "consent"> & {
+  consent: Record<string, boolean | string>;
+};
+
+export interface GtmOptions {
+  includeNotApplicable?: boolean;
+  asStringValues?: boolean;
+}
```

Comment:
@tvandort feel free to add suggestions for what the options should ultimately look like here.

Reply:
IMO: 
```typescript
export interface GtmOptions {
  nonApplicableFlagMode: "omit" | "include";
  flagType?: "boolean" | "string";
}
```

Couple of reasons for this...

1. future extensibility
2. clarity in communication
3. potential for non-valid state expression in the future

This article goes over some of these concerns. I've definitely read other arguments before but I can't find any other compelling ones that I'm willing to link to.
https://katafrakt.me/2024/11/09/booleans-are-a-trap/

Reply:
I have no issue with this concept at all, I think it makes a lot of sense. One nit-pick though, would it make sense to have the 2nd option of `flagType` to be "mechanism" instead of the generic "string" to leave room for other string uses in the future?

Reply:
After talking it out Jason and I settled on `flagType: "consentMechanism"` since that is the subset of all strings for which the `Fides.consent` object's flag type changes to in this new mode.